src/concrete/ml/quantization/quantized_module.py

```diff
@@ -99,28 +99,39 @@ def __init__(
         # Set base attributes for API consistency. This could be avoided if an abstract base class
         # is created for both Concrete ML models and QuantizedModule
         # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/2899
-        self.fhe_circuit = None
+
+        all_or_none_params = [
+            ordered_module_input_names,
+            ordered_module_output_names,
+            quant_layers_dict,
+        ]
+        assert_true(
+            all(v is None or v == {} for v in all_or_none_params)
+            or not any(v is None or v == {} for v in all_or_none_params),
+            "All of ordered_module_input_names, ordered_module_output_names, "
```

Comment:
I would rephrase this by "Please either set all three 'ordered_module_input_names', 'ordered_module_output_names' and 'quant_layers_dict' or none of them" 

also, we should avoid putting `assert` or `assert_true` for user inputs like this and instead add ValueError raises + an associate small test (cf https://github.com/zama-ai/concrete-ml-internal/issues/4232)

Reply:
Sure I rephrased.

About the second part, it's not really user inputs here. Users use the QuantizedModule through the PTQ. I can add a test anyway yes.

Reply:
it's still a public class that we even can serialize like any built-in models. But actually I meant "inputs" more than 'user inputs", asserts should not really be used to check input value formats